File: tests/output/php/src/methods/requests/AbtestingTest.php
<?php

namespace Algolia\AlgoliaSearch\Test\Api;

use Algolia\AlgoliaSearch\Api\AbtestingApi;
use Algolia\AlgoliaSearch\Configuration\AbtestingConfig;
use Algolia\AlgoliaSearch\Http\HttpClientInterface;
use Algolia\AlgoliaSearch\Http\Psr7\Response;
use Algolia\AlgoliaSearch\RetryStrategy\ApiWrapper;
use Algolia\AlgoliaSearch\RetryStrategy\ClusterHosts;
use PHPUnit\Framework\TestCase;
use Psr\Http\Message\RequestInterface;

/**
 * AbtestingTest
 *
 * @category Class
 * @package  Algolia\AlgoliaSearch
 */
class AbtestingTest extends TestCase implements HttpClientInterface
{
    /**
     * @var RequestInterface[]
     */
    private $recordedRequests = [];

    protected function assertRequests(array $requests)
    {
        $this->assertGreaterThan(0, count($requests));
        $this->assertEquals(count($requests), count($this->recordedRequests));

        foreach ($requests as $i => $request) {
            $recordedRequest = $this->recordedRequests[$i];

            $this->assertEquals($request['method'], $recordedRequest->getMethod());
            $this->assertEquals($request['path'], $recordedRequest->getUri()->getPath());
            $this->assertEquals($request['body'], $recordedRequest->getBody()->getContents());
        }
    }

    public function sendRequest(RequestInterface $request, $timeout, $connectTimeout)
    {
        $this->recordedRequests[] = $request;

        return new Response(200, [], '{}');
    }

    protected function getClient()
    {
        $api = new ApiWrapper($this, AbtestingConfig::create(), ClusterHosts::create('127.0.0.1'));
        $config = AbtestingConfig::create('foo', 'bar');

        return new AbtestingApi($api, $config);
    }

    /**
     * Test case for addABTests
     * AddABTests with minimal parameters
     */
    public function testAddABTests0()
    {
        $client = $this->getClient();
        $params = json_decode('{"endAt":"2022-12-31T00:00:00.000Z","name":"myABTest","variant":[{"index":"AB_TEST_1","trafficPercentage":30},{"index":"AB_TEST_2","trafficPercentage":50}]}', true);

        $client->addABTests($params);

        $this->assertRequests([
            [
                'path' => '/2/abtests',
                'method' => 'POST',
                'body' => '{"endAt":"2022-12-31T00:00:00.000Z","name":"myABTest","variant":[{"index":"AB_TEST_1","trafficPercentage":30},{"index":"AB_TEST_2","trafficPercentage":50}]}',
            ],
        ]);
    }

    /**
     * Test case for deleteABTest
     * DeleteABTest
     */
    public function testDeleteABTest0()
    {
        $client = $this->getClient();
        $params = json_decode('{"id":42}', true);

        $client->deleteABTest($params);

        $this->assertRequests([
            [
                'path' => '/2/abtests/42',
                'method' => 'DELETE',
                'body' => '',
            ],
        ]);
    }

    /**
     * Test case for getABTest
     * GetABTest
     */
    public function testGetABTest0()
    {
        $client = $this->getClient();
        $params = json_decode('{"id":42}', true);

        $client->getABTest($params);

        $this->assertRequests([
            [
                'path' => '/2/abtests/42',
                'method' => 'GET',
                'body' => '',
            ],
        ]);
    }

    /**
     * Test case for listABTests
     * ListABTests with minimal parameters
     */
    public function testListABTests0()
    {
        $client = $this->getClient();
        $params = json_decode('{"offset":42,"limit":21}', true);

        $client->listABTests($params);

        $this->assertRequests([
            [
                'path' => '/2/abtests',
                'method' => 'GET',
                'body' => '',
            ],
        ]);
    }

    /**
     * Test case for stopABTest
     * StopABTest
     */
    public function testStopABTest0()
    {
        $client = $this->getClient();
        $params = json_decode('{"id":42}', true);

        $client->stopABTest($params);

        $this->assertRequests([
            [
                'path' => '/2/abtests/42/stop',
                'method' => 'POST',
                'body' => '',
            ],
        ]);
    }
}

File: tests/output/java/src/test/java/com/algolia/methods/requests/abtesting.test.java
package com.algolia.methods.requests;

import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
import static org.junit.jupiter.api.Assertions.assertEquals;

import com.algolia.JSON;
import com.algolia.Pair;
import com.algolia.api.AbtestingApi;
import com.algolia.model.abtesting.*;
import com.algolia.utils.echo.*;
import com.google.gson.reflect.TypeToken;
import java.util.*;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.TestInstance;
import org.skyscreamer.jsonassert.JSONAssert;
import org.skyscreamer.jsonassert.JSONCompareMode;

@TestInstance(TestInstance.Lifecycle.PER_CLASS)
class AbtestingApiTests {

  private AbtestingApi client;

  @BeforeAll
  void init() {
    client = new AbtestingApi("appId", "apiKey", new EchoRequester());
  }

  @Test
  @DisplayName("addABTests with minimal parameters")
  void addABTestsTest0() {
    AddABTestsRequest addABTestsRequest0 = new AddABTestsRequest();
    {
      String endAt1 = "2022-12-31T00:00:00.000Z";

      addABTestsRequest0.setEndAt(endAt1);
      String name1 = "myABTest";

      addABTestsRequest0.setName(name1);

      List<AddABTestsVariant> variant1 = new ArrayList<>();

      {
        AbTestsVariant variant_02 = new AbTestsVariant();
        {
          String index3 = "AB_TEST_1";

          variant_02.setIndex(index3);

          int trafficPercentage3 = 30;

          variant_02.setTrafficPercentage(trafficPercentage3);
        }
        variant1.add(AddABTestsVariant.ofAbTestsVariant(variant_02));

        AbTestsVariant variant_12 = new AbTestsVariant();
        {
          String index3 = "AB_TEST_2";

          variant_12.setIndex(index3);

          int trafficPercentage3 = 50;

          variant_12.setTrafficPercentage(trafficPercentage3);
        }
        variant1.add(AddABTestsVariant.ofAbTestsVariant(variant_12));
      }
      addABTestsRequest0.setVariant(variant1);
    }

    EchoResponseInterface req = (EchoResponseInterface) assertDoesNotThrow(() -> {
        return client.addABTests(addABTestsRequest0);
      }
    );

    assertEquals(req.getPath(), "/2/abtests");
    assertEquals(req.getMethod(), "POST");

    assertDoesNotThrow(() -> {
      JSONAssert.assertEquals(
        "{\"endAt\":\"2022-12-31T00:00:00.000Z\",\"name\":\"myABTest\",\"variant\":[{\"index\":\"AB_TEST_1\",\"trafficPercentage\":30},{\"index\":\"AB_TEST_2\",\"trafficPercentage\":50}]}",
        req.getBody(),
        JSONCompareMode.STRICT_ORDER
      );
    });
  }

  @Test
  @DisplayName("deleteABTest")
  void deleteABTestTest0() {
    int id0 = 42;

    EchoResponseInterface req = (EchoResponseInterface) assertDoesNotThrow(() -> {
        return client.deleteABTest(id0);
      }
    );

    assertEquals(req.getPath(), "/2/abtests/42");
    assertEquals(req.getMethod(), "DELETE");
  }

  @Test
  @DisplayName("getABTest")
  void getABTestTest0() {
    int id0 = 42;

    EchoResponseInterface req = (EchoResponseInterface) assertDoesNotThrow(() -> {
        return client.getABTest(id0);
      }
    );

    assertEquals(req.getPath(), "/2/abtests/42");
    assertEquals(req.getMethod(), "GET");
  }

  @Test
  @DisplayName("listABTests with minimal parameters")
  void listABTestsTest0() {
    int offset0 = 42;

    int limit0 = 21;

    EchoResponseInterface req = (EchoResponseInterface) assertDoesNotThrow(() -> {
        return client.listABTests(offset0, limit0);
      }
    );

    assertEquals(req.getPath(), "/2/abtests");
    assertEquals(req.getMethod(), "GET");

    HashMap<String, String> expectedQuery = JSON.deserialize(
      "{\"offset\":\"42\",\"limit\":\"21\"}",
      new TypeToken<HashMap<String, String>>() {}.getType()
    );
    List<Pair> acutalQuery = req.getQueryParams();
    for (Pair p : acutalQuery) {
      assertEquals(expectedQuery.get(p.getName()), p.getValue());
    }
  }

  @Test
  @DisplayName("stopABTest")
  void stopABTestTest0() {
    int id0 = 42;

    EchoResponseInterface req = (EchoResponseInterface) assertDoesNotThrow(() -> {
        return client.stopABTest(id0);
      }
    );

    assertEquals(req.getPath(), "/2/abtests/42/stop");
    assertEquals(req.getMethod(), "POST");
  }
}
